src/metadata/metadata.rs

```diff
@@ -0,0 +1,207 @@
+use chrono::{Datelike, NaiveDate};
+use regex::Regex;
+use titlecase::titlecase;
+
+use super::join::Joinable;
+use super::parse;
+use crate::config::SHORT_TO_LONG_STATUS;
+use crate::line::Line;
+use crate::spec::Spec;
+
+pub type Date = NaiveDate;
+
+#[derive(Debug, Default, Clone)]
+pub struct MetadataManager {
+    pub has_metadata: bool,
+    pub abs: Option<Vec<String>>,
+    pub canonical_url: Option<String>,
+    pub date: Option<Date>,
+    pub ed: Option<String>,
+    pub editors: Option<Vec<String>>,
+    pub group: Option<String>,
+    pub level: Option<String>,
+    pub shortname: Option<String>,
+    pub raw_status: Option<String>,
+    pub title: Option<String>,
+}
+
+impl MetadataManager {
+    pub fn new() -> MetadataManager {
+        MetadataManager {
+            has_metadata: false,
```

Comment:
This is not needed, `Default` for `bool` is just false, so this can just be `Self::default()`.